srcts/src/shiny/shinyapp.ts

```diff
@@ -1053,6 +1053,12 @@ class ShinyApp {
         const $tabContent = getTabContent($tabset);
         let tabsetId = $parentTabset.attr("data-tabsetid");
 
+        // TODO: Only render tab content HTML once
+        // The following lines turn the content/nav control HTML into DOM nodes,
+        // but we don't insert these directly, instead we take the HTML from
+        // these nodes and pass it through `renderContentAsync()`. This means
+        // the inserted HTML may not perfectly match the message HTML, esp. if
+        // the content uses web components that alter their HTML when loaded.
         const $divTag = $(message.divTag.html);
```

Comment:
Maybe this shouldn't be a comment in the source, but the idea is to come back to this in a subsequent PR and see if we can do this with only one `html -> DOM` pass instead of two.